### PR TITLE
fix(curriculum): changed test text to use should for Coding Interview Prep - part 1 of 2

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Related to PR #36860 

This PR makes sure all challenges' tests in the following sections of the `Coding Interview Prep` section use "should " or "should not".

* Algorithms
* Data Structures

### CODE REVIEW

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/check-if-an-element-is-present-in-a-binary-search-tree.english.md [hunk 1]
```diff
@@ -23,13 +23,13 @@ In this challenge, we will create a utility for our tree. Write a method <code>i
...
-  - text: The binary search tree has a method called <code>isPresent</code>.
+  - text: The binary search tree should have a method called <code>isPresent</code>.
     testString: assert((function() { var test = false; if (typeof BinarySearchTree !== 'undefined') { test = new BinarySearchTree() } else { return false; }; return (typeof test.isPresent == 'function')})());
-  - text: The <code>isPresent</code> method correctly checks for the presence or absence of elements added to the tree.
+  - text: The <code>isPresent</code> method should correctly check for the presence or absence of elements added to the tree.
     testString: assert((function() { var test = false; if (typeof BinarySearchTree !== 'undefined') { test = new BinarySearchTree() } else { return false; }; if (typeof test.isPresent !== 'function') { return false; }; test.add(4); test.add(7); test.add(411); test.add(452); return ( test.isPresent(452) && test.isPresent(411) && test.isPresent(7) && !test.isPresent(100) ); })());
-  - text: <code>isPresent</code> handles cases where the tree is empty.
+  - text: <code>isPresent</code> should handles cases where the tree is empty.
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/check-if-an-element-is-present-in-a-binary-search-tree.english.md]:
```suggestion
  - text: <code>isPresent</code> should handle cases where the tree is empty.
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md [hunk 2]
```diff
@@ -23,21 +23,21 @@ Be sure to write your code to account for collisions!
...
 ```yml
 tests:
-  - text: The HashTable data structure exists.
+  - text: The HashTable data structure should exist.
     testString: assert((function() { var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; return (typeof test === 'object')})());
-  - text: The HashTable has an add method.
+  - text: The HashTable should have an add method.
     testString: assert((function() { var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; return ((typeof test.add) === 'function')})());
-  - text: The HashTable has an remove method.
+  - text: The HashTable should have an remove method.
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md]:
```suggestion
  - text: The HashTable should have a remove method.
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md [hunk 3]
```diff
@@ -23,21 +23,21 @@ Be sure to write your code to account for collisions!
...
     testString: assert((function() { var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; return ((typeof test.remove) === 'function')})());
-  - text: The HashTable has an lookup method.
+  - text: The HashTable should have an lookup method.
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md]:
```suggestion
  - text: The HashTable should have a lookup method.
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md [hunk 4]
```diff
@@ -23,23 +23,23 @@ Let's finish our <code>remove</code> method by handling the third case. We've pr
...
+  - text: The <code>remove</code> method should remove nodes with one child.
     testString: assert((function() { var test = false; if (typeof BinarySearchTree !== 'undefined') { test = new BinarySearchTree() } else { return false; }; if (typeof test.remove !== 'function') { return false; }; test.add(-1); test.add(3); test.add(7); test.add(16); test.remove(16); test.remove(7); test.remove(3); return (test.inorder().join('') == '-1'); })());
-  - text: Removing the root in a tree with two nodes sets the second to be the root.
+  - text: Removing the root in a tree with two nodes should set the second to be the root.
     testString: assert((function() { var test = false; if (typeof BinarySearchTree !== 'undefined') { test = new BinarySearchTree() } else { return false; }; if (typeof test.remove !== 'function') { return false; }; test.add(15); test.add(27); test.remove(15); return (test.inorder().join('') == '27'); })());
-  - text: The <code>remove</code> method removes nodes with two children while maintaining the binary search tree structure.
+  - text: The <code>remove</code> method should removes nodes with two children while maintaining the binary search tree structure.
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md]:
```suggestion
  - text: The <code>remove</code> method should remove nodes with two children while maintaining the binary search tree structure.
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md [hunk 5]
```diff
@@ -23,23 +23,23 @@ Let's finish our <code>remove</code> method by handling the third case. We've pr
...
     testString: assert((function() { var test = false; if (typeof BinarySearchTree !== 'undefined') { test = new BinarySearchTree() } else { return false; }; if (typeof test.remove !== 'function') { return false; }; test.add(1); test.add(4); test.add(3); test.add(7); test.add(9); test.add(11); test.add(14); test.add(15); test.add(19); test.add(50); test.remove(9); if (!test.isBinarySearchTree()) { return false; }; test.remove(11); if (!test.isBinarySearchTree()) { return false; }; test.remove(14); if (!test.isBinarySearchTree()) { return false; }; test.remove(19); if (!test.isBinarySearchTree()) { return false; }; test.remove(3); if (!test.isBinarySearchTree()) { return false; }; test.remove(50); if (!test.isBinarySearchTree()) { return false; }; test.remove(15); if (!test.isBinarySearchTree()) { return false; }; return (test.inorder().join('') == '147'); })());
-  - text: The root can be removed on a tree of three nodes.
+  - text: The root should be able to removable on a tree of three nodes.
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md]:
```suggestion
  - text: The root should be able to be removed on a tree of three nodes.
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/use-spread-and-notes-for-es5-set-integration.english.md [hunk 6]
```diff
@@ -30,7 +30,7 @@ Now you've successfully learned how to use the ES6 <code>Set()</code> object, go
 
 ```yml
 tests:
-  - text: Your Set was returned correctly!
+  - text: <code>checkSet(new Set([1,2,3,4,5,6,7])</code> should return <code>[1, 2, 3, 4, 5, 6, 7]</code>!
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/use-spread-and-notes-for-es5-set-integration.english.md]:
```suggestion
  - text: <code>checkSet(new Set([1,2,3,4,5,6,7])</code> should return <code>[1, 2, 3, 4, 5, 6, 7]</code>.
```

**Review by moT01 — CHANGES_REQUESTED** [review]
Couple minor things.

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md (hunk 5 above)
Comment (RandellDawson) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md]: @moT01 I think what I meant to change it to was:
> The root should be removable on a tree of three nodes.

If you are ok with that, can you change the suggestion?

Comment (moT01) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md]: ✔️

**Comment by RandellDawson** [comment]
@moT01 OK, I think that takes care of all the suggested changes for this one.  Thanks for the review.  I noticed many of the typos were already present before my changes.

**Review by moT01 — APPROVED** [review]
Yea, I saw that some of them were preexisting, but we're right here - might as well fix it quick when it's that noticeable and easy.

🎉  LGTM 🎉

**Review by Manish-Giri — COMMENTED** [review]
LGTM, but while we're at it, I noticed something on the "create a map" challenge - 

> The clear method empties the map and the size method returns the number of items present in the map.

Doesn't the language sound contradictory? If the map is emptied, why would there be any items present in the map? Might be a good idea to change the text in second part of the instruction.

**Comment by moT01** [comment]
Seems like the text just explains what the method does, not that it actually ran that method or emptied the map.

I think that could probably be split up into two tests, but I would maybe make it a separate issue - or just leave it

**Review by Manish-Giri — APPROVED** [review]
All right, I'll create a separate PR to handle that case.

**Pull request merged**